Tests/LibCompress/TestDeflate.cpp

```diff
@@ -136,43 +131,27 @@ TEST_CASE(deflate_round_trip_compress)
     auto original = ByteBuffer::create_zeroed(2048).release_value();
     fill_with_random(original.bytes().trim(1024)); // we pre-filled the second half with 0s to make sure we test back references as well
     // Since the different levels just change how much time is spent looking for better matches, just use fast here to reduce test time
-    auto compressed = TRY_OR_FAIL(Compress::DeflateCompressor::compress_all(original, Compress::DeflateCompressor::CompressionLevel::FAST));
+    auto compressed = TRY_OR_FAIL(Compress::DeflateCompressor::compress_all(original));
     auto uncompressed = TRY_OR_FAIL(Compress::DeflateDecompressor::decompress_all(compressed));
     EXPECT(uncompressed == original);
 }
 
 TEST_CASE(deflate_round_trip_compress_large)
 {
-    auto size = Compress::DeflateCompressor::block_size * 2;
-    auto original = ByteBuffer::create_uninitialized(size).release_value(); // Compress a buffer larger than the maximum block size to test the sliding window mechanism
+    // Compress a buffer larger than the maximum block size to test the sliding window mechanism
+    auto original = TRY_OR_FAIL(ByteBuffer::create_uninitialized((32 * KiB - 1) * 2));
     fill_with_random(original);
+
     // Since the different levels just change how much time is spent looking for better matches, just use fast here to reduce test time
-    auto compressed = TRY_OR_FAIL(Compress::DeflateCompressor::compress_all(original, Compress::DeflateCompressor::CompressionLevel::FAST));
+    auto compressed = TRY_OR_FAIL(Compress::DeflateCompressor::compress_all(original));
     auto uncompressed = TRY_OR_FAIL(Compress::DeflateDecompressor::decompress_all(compressed));
+    EXPECT(uncompressed.span() == original.span().slice(0, uncompressed.size()));
     EXPECT(uncompressed == original);
 }
 
 TEST_CASE(deflate_compress_literals)
 {
     // This byte array is known to not produce any back references with our lz77 implementation even at the highest compression settings
     Array<u8, 0x13> test { 0, 0, 0, 0, 0x72, 0, 0, 0xee, 0, 0, 0, 0x26, 0, 0, 0, 0x28, 0, 0, 0x72 };
-    auto compressed = TRY_OR_FAIL(Compress::DeflateCompressor::compress_all(test, Compress::DeflateCompressor::CompressionLevel::GOOD));
-}
-
-TEST_CASE(ossfuzz_63183)
-{
-    auto path = TEST_INPUT("clusterfuzz-testcase-minimized-FuzzDeflateCompression-6163230961303552.fuzz"sv);
-    auto test_file = MUST(Core::File::open(path, Core::File::OpenMode::Read));
-    auto test_data = MUST(test_file->read_until_eof());
-    auto compressed = TRY_OR_FAIL(Compress::DeflateCompressor::compress_all(test_data, Compress::DeflateCompressor::CompressionLevel::GOOD));
-    auto decompressed = TRY_OR_FAIL(Compress::DeflateDecompressor::decompress_all(compressed));
-    EXPECT(test_data == decompressed);
-}
```

Comment:
Any specific reason to remove these cases? I understand that they were tightly coupled to our implementation, but maybe mention it in the commit message.

Reply:
These were added in b7491675062cebd3bad2e56bfe8ba5c9790944bd and 75bd1308c52555624cba282fa4c8c8489955ba76 to check for crashes. Having changed the implementation substantially they lose their original purpose.

I can add them back in or mention the above in the commit message.

Reply:
Yeah just mentioning it is fine imo.